dataikuapi/fm/instances.py

```diff
@@ -37,14 +37,12 @@ def with_dss_node_type(self, dss_node_type):
         """
         Set the DSS Node type of the instance to create
 
-        :param str dss_node_type: Optional , the type of the dss node to create. Supports "design", "automation ordeployer". Defaults to "design"
+        :param dss_node_type: Optional , the type of the dss node to create. Defaults to `DESIGN`.
+        :type dss_node_type: :class:`dataikuapi.fm.instances.FMNodeType`
         :rtype: :class:`dataikuapi.fm.instances.FMInstanceCreator`
         """
-        if dss_node_type not in ["design", "automation", "deployer"]:
-            raise ValueError(
-                'Only "design", "automation" or "deployer" dss_node_type are supported'
-            )
-        self.data["dssNodeType"] = dss_node_type
+
+        self.data["dssNodeType"] = dss_node_type.value
```

Comment:
to complement the comment from @spicquenot , I'd maintain backwards compatibility by doing something like

```
if isinstance(dss_node_type, enum):
    self.data["dssNodeType"] = dss_node_type.value
else:
    self.data["dssNodeType"] = dss_node_type
```

(or something that is python2 AND python3 compatible)

the signature and comment change is fine IMHO

Reply:
BW compatibility added